### PR TITLE
Issue 1627 - Save fails when a single resource is used with multiple related resources

### PR DESCRIPTION
**Pull request by blayzen-w**
See issue [1627](https://github.com/django-tastypie/django-tastypie/issues/1627) for details

### CODE REVIEW

**Comment by georgedorn** [comment]
Closing and re-opening to convince github actions to run the tests.

File: tastypie/resources.py [hunk 1]
```diff
@@ -2406,6 +2406,7 @@ def save(self, bundle, skip_errors=False):
 
         if obj_id not in bundle.objects_saved or bundle.obj._state.adding:
             bundle.obj.save()
+            obj_id = self.create_identifier(bundle.obj)
```
Comment (georgedorn) [inline tastypie/resources.py:2409]: I don't understand how this one line makes a difference; the `obj_id` is already created on line 2405.

Comment (blayzen-w) [reply tastypie/resources.py:2409]: When the first create_identifier function is called the object has not been saved and does not have a id. This makes the return value of that function always have None at the end (ex: `core.mycontaineritemmodel.None`) which is the same as every other unsaved object which causes the saving issue. Once the object is saved the create identifier function will return a unique value.

Comment (georgedorn) [reply tastypie/resources.py:2409]: Ah, I see now; on the first call, the PK is `None`, so the `obj_id` ends up shadowing all other not-yet-saved objects.

**Comment by georgedorn** [comment]
The fix makes sense now and the test correctly fails when the fix is not applied.  Style check is failing (see the flake8 step in any of the test matrix entries) but I'll merge once that's been fixed.

**Comment by blayzen-w** [comment]
Thanks for taking a look at it. I'll start working on fixing the code style issues in a bit.

**Comment by georgedorn** [comment]
Thanks for the PR; sorry I didn't get to it sooner, but the Travis CI debacle meant no test matrix system and it took several hours of effort to migrate away from it.  But I absolutely appreciate the fix + tests; it makes it a lot easier to be confident in the fix.

**Pull request merged**